cmd/mps-control-daemon/mount/mount-shm.go

```diff
@@ -54,11 +74,58 @@ func mountShm(c *cli.Context) error {
 		return fmt.Errorf("error creating directory %v: %w", shmDir, err)
 	}
 
-	//  TODO: What should the size of the shm be
-	mountOptions := []string{"rw", "nosuid", "nodev", "noexec", "relatime", "size=65536k"}
+	sizeArg := fmt.Sprintf("size=%v", opts.getShmSize())
+	mountOptions := []string{"rw", "nosuid", "nodev", "noexec", "relatime", sizeArg}
 	if err := mounter.Mount("shm", shmDir, "tmpfs", mountOptions); err != nil {
 		return fmt.Errorf("error mounting %v as tmpfs: %w", shmDir, err)
 	}
 
 	return nil
 }
+
+// getShmSize returns the default shm size.
+// This reads /proc/meminfo to get the total memory to calculate this.
+func (o *options) getShmSize() (size string) {
+	defer func() {
+		if size == "" {
+			size = "65536k"
+		}
+		klog.InfoS("Calculated /dev/shm size", "size", size)
+	}()
```

Comment:
Why this instead of:
```
fallbackSize := "65536k"
```
And then explicitly return this on errors instead of `""`

Reply:
Sure.

Reply:
Updated.